kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/impl/cache/Reflector.java

```diff
@@ -171,6 +182,12 @@ public boolean isWatching() {
 
   class ReflectorWatcher implements Watcher<T> {
 
+    private final Supplier<InformerExceptionHandler> exceptionHandlerSupplier;
+
+    ReflectorWatcher(Supplier<InformerExceptionHandler> exceptionHandlerSupplier) {
```

Comment:
Is using a Supplier needed to get around needing to pass a final reference to closures?

Reply:
I was just trying to avoid the exceptional case of setting the handler after starting.  I'll simplify that more.